### ISSUE TITLE
BibTeX for software records

### ISSUE THREAD

**Issue by ehenneken** [adsabs/export_service#77]
To comply with the AAS author guide, our BibTeX output for software records should look like
```
@misc{2015zndo.soft17920F,
author       = {{Kerzendorf}, W. and {Do}, T.},
title        = {{Starkit: second release}},
month        = aug,
year         = 2015,
doi          = {10.5281/zenodo.1117920},
version      = {0.3},
publisher    = {Zenodo}
}
```
Currently we produce the following BibTeX:
```
@MISC{2015zndo.soft17920F,
author = {{Kerzendorf}, W. and {Do}, T.},
title = "{Starkit: second release}",
howpublished = {Zenodo Software Release, 2015},
year = 2015,
month = aug,
abstract = "{Toolkit to model stellar spectroscopy and photometry
}",
doi = {10.5281/zenodo.1117920},
adsurl = {http://adsabs.harvard.edu/abs/2015zndo.soft17920F},
adsnote = {Provided by the SAO/NASA Astrophysics Data System}
}
```
So, in practice this means the following:

1. The `howpublished` field should be replaced with `publisher` and only say `Zenodo` (the Solr documents will contain that `publisher` field)
2. There needs to be a `version` field, populated with the value of the equivalent field in the Solr document for the software record.

**Comment by golnazads** [comment]
we still do not have `version` data in solr.

**Comment by golnazads** [comment]
When are we expecting to have some data in solr for this. Just realized that publisher is also not populated? @ehenneken @aaccomazzi @marblestation

**Comment by ehenneken** [comment]
If all goes according to plan, data for this should go into Solr before the end of this year.

**Comment by marblestation** [comment]
@ehenneken, from your explanation I interpret that export should return `publisher = "Zenodo"` for software records but this sounds like something hard-coded in the microservice, while I would expect the microservice to grab this information from Solr and return whatever is there (or not include the field if Solr does not contain info yet). Could you clarify what you meant?

**Comment by aaccomazzi** [comment]
I think we should keep the export code as simple as possible and do this:

1. if the SOLR `publisher`field exist, output it in the bibtex `publisher` field
2. if the SOLR `version`field exist, output it in the bibtex `version` field

No other special treatment or business logic in the export service.

BTW the inclusion of these two fields comes mostly from feedback from the AAS (@augustfly), I find plenty of inconsistent advice for `misc` entries out there in the wild, including using `howpublished` as a general catch-all field.  This is true even for biblatex, it seems

**Comment by golnazads** [comment]
what about pages and eid, for example

@MISC{2017ascl.soft06009C,
       author = {{Casey}, Andrew R.},
        title = "{sick: Spectroscopic inference crank}",
     keywords = {Software},
 howpublished = {Astrophysics Source Code Library},
         year = 2017,
        month = Jun,
          eid = {ascl:1706.009},
        pages = {ascl:1706.009},
archivePrefix = {ascl},
       eprint = {1706.009},
       adsurl = {https://ui.adsabs.harvard.edu/#abs/2017ascl.soft06009C},
      adsnote = {Provided by the SAO/NASA Astrophysics Data System}
}


      {
        "bibcode":"2017ascl.soft06009C",
        "doctype":"software",
        "eid":"ascl:1706.009"}

**Comment by golnazads** [comment]
because of the latest fix to identifications all bibstem:ascl would also have archivePrefix and eprint. should these be included for software?

**Comment by ehenneken** [comment]
It is correct to call this “software”, but there is a subtle difference with records with the bibstem “zndo.soft”: ASCL does not distinguish versions. The “pages” field does not make any sense. Just like Zenodo records, ASCL records are basically just metadata holders. I would not populate the “pages” field.

**Comment by golnazads** [comment]
Hi. For bibstem:ascl we have for example
{ "page":["ascl:9903.001"], "doctype":"software", "bibcode":
"1999ascl.soft03001N", "eid":"ascl:9903.001"}, { "page":["ascl:9905.001"], "
bibcode":"1999ascl.soft05001N", "doctype":"software", "eid":"ascl:9905.001"},
{ "page":["ascl:9906.001"], "doctype":"software", "bibcode":
"1999ascl.soft06001F", "eid":"ascl:9906.001"},
so page and eid is the same for these. I removed page, but not eid. Is that
how you want it?
For arXiv is the samething:
{ "page":["cs/0401005"], "bibcode":"2004cs........1005V", "doctype":"eprint",
"eid":"cs/0401005"}, { "page":["astro-ph/0611456"], "doctype":"eprint", "
bibcode":"2006astro.ph.11456F", "eid":"astro-ph/0611456"}, { "page":[
"astro-ph/0603409"], "doctype":"eprint", "bibcode":"2006astro.ph..3409P", "
eid":"astro-ph/0603409"},

**Comment by aaccomazzi** [comment]
I would keep both page and eid

**Comment by golnazads** [comment]
ok. thank you so much.

**Comment by golnazads** [comment]
implemented in v2.0.34.
